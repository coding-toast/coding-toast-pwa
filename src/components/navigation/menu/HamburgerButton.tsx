import * as React from 'react'
import HamburgerIcon from '../../../images/hamburger.png'
import Menu from './Menu'
import HamburgerStyles from './HamburgerButton.module.scss'

const HamburgerButton: React.FC = () => {
  const [open, setIsOpen] = React.useState(false)
  return (
    <div>
      <button type="button" className={HamburgerStyles.hamburgerButton} onClick={() => setIsOpen(!open)} onKeyDown={() => setIsOpen(!open)}>
        <img src={HamburgerIcon} className={HamburgerStyles.hamburgerIcon} alt="Open Menu" />
      </button>
      {open ? <Menu /> : ''}
    </div>
  )
}

export default HamburgerButton
