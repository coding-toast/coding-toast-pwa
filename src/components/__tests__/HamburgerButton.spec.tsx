import React from 'react'
import renderer, { act } from 'react-test-renderer'

import HamburgerButton from '../navigation/menu/HamburgerButton'

describe('HamburgerButton', () => {
  it('renders correctly', () => {
    const component = renderer.create(<HamburgerButton />).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('renders opened correctly', () => {
    const component = renderer.create(<HamburgerButton />)

    act(() => {
      component.root.findByType('button').props.onClick()
    })

    expect(component.toJSON()).toMatchSnapshot()
  })
})
