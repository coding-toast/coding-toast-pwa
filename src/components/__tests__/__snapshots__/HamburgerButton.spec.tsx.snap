// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerButton renders correctly 1`] = `
<div>
  <button
    className="hamburgerButton"
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "background": "none",
        "border": "none",
        "color": "inherit",
        "cursor": "pointer",
        "font": "inherit",
        "outline": "inherit",
        "padding": 0,
      }
    }
    type="button"
  >
    <img
      alt="Open Menu"
      src="test-file-stub"
      style={
        Object {
          "color": "#70F3FF",
          "display": "block",
          "padding": "0",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </button>
  
</div>
`;

exports[`HamburgerButton renders opened correctly 1`] = `
<div>
  <button
    className="hamburgerButton"
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "background": "none",
        "border": "none",
        "color": "inherit",
        "cursor": "pointer",
        "font": "inherit",
        "outline": "inherit",
        "padding": 0,
      }
    }
    type="button"
  >
    <img
      alt="Open Menu"
      src="test-file-stub"
      style={
        Object {
          "color": "#70F3FF",
          "display": "block",
          "padding": "0",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </button>
  <div>
    <div
      id="menuLinks"
    >
      <a
        href="/"
        style={
          Object {
            "color": "white",
            "fontSize": "24px",
            "padding": "10px",
            "textDecoration": "none",
          }
        }
      >
        About
      </a>
      <a
        href="/"
        style={
          Object {
            "color": "white",
            "fontSize": "24px",
            "padding": "10px",
            "textDecoration": "none",
          }
        }
      >
        Contact
      </a>
      <a
        href="/"
        style={
          Object {
            "color": "white",
            "fontSize": "24px",
            "padding": "10px",
            "textDecoration": "none",
          }
        }
      >
        Login
      </a>
    </div>
  </div>
</div>
`;
